src/main/target/MATEKF405/target.h

```diff
@@ -118,11 +118,11 @@
 #define UART5_RX_PIN            PD2
 #define UART5_TX_PIN            PC12
 
-//#define USE_SOFTSERIAL1
-//#define SOFTSERIAL_1_RX_PIN      PA1  //RX4
-//#define SOFTSERIAL_1_TX_PIN      PA0  //TX4
+#define USE_SOFTSERIAL1
+#define SOFTSERIAL_1_RX_PIN      PA1  //RX4
+#define SOFTSERIAL_1_TX_PIN      PA0  //TX4
```

Comment:
There is a clash between `USE_SOFTSERIAL1` and `USE_UART4` - they use the same pins. If softserial is enabled UART4 should be removed from the list of UARTs on this target.

Reply:
I think the goal was to keep the possibility to use one or the other. Can't the two coexist meaning if nothing is configured on UART4 enabling SS1 and using it instead would work ?

Reply:
I'm worried about the scenario when user will configure both. This needs to be prevented.

Reply:
A solution could be to add targets replacing UART4 with softserial. MATEKF405_1SS and MATEKF405OSD_1SS.

Reply:
Yes, Use one or the other. UART4 can work as normal UART if users don't enable softserial in configurator.

Reply:
@shellixyz I think no need to add more targets. Now it is a good solution. . 5xUarts are still available for non-Frsky users. 4xUarts+1xSoftserial for Frsky users after enable softserial.  We will add this guide in our FC page.  in BF there are some targets use this type of hack on UART pins also.

Reply:
BTW, F4 telemetry_inversion = OFF by default.  F722 telemetry_inversion = OFF default also. Need to set them = ON to get Frsky telemetry working.  can telemetry_inversion be defined as ON by default in firmware.

Reply:
@Ogocology Since you want to use soft serial for smartport I guess you connected your receiver without the uninversion hack so your issue is probably that you need `telemetry_inverted = OFF`. It means normal expected inversion. So for smartport it means inverted.

Reply:
Its working now with telemetry_inverted = OFF.
I also got servo working on s6 pad for camera tilt servo.

I remain grateful to all who consistently strive to make this level of support available (and FREE).

Cheers!